File: givewell-meetings/generate_summaries.py
#!/usr/bin/env python3
"""
Generate comprehensive 1000-word summaries for all GiveWell board meeting episodes.
"""

import json
import anthropic
import os
from typing import Dict, Any

def generate_summary(episode: Dict[str, Any], client) -> str:
    """Generate a comprehensive ~1000-word summary for an episode."""

    # Combine all attachment texts
    attachment_texts = []
    for att in episode.get('attachments', []):
        if 'text' in att and att['text']:
            attachment_texts.append(f"=== {att.get('title', 'Document')} ===\n{att['text']}\n")

    combined_text = "\n\n".join(attachment_texts)

    prompt = f"""Based on the following board meeting documents for a {episode['title']} held on {episode['dateDisplay']}, generate a comprehensive ~1000-word summary that acts as a professional brief.

The summary should cover:
- Key decisions made and votes taken
- Important discussions and debates
- Financial/budget information with specific numbers
- Strategic direction or policy changes
- Personnel or organizational changes
- Any significant outcomes or action items

Keep the tone professional and factual. Focus on substance over process. Be specific with numbers, names, and decisions.

Meeting Documents:
{combined_text}

Generate a comprehensive summary (approximately 1000 words):"""

    message = client.messages.create(
        model="claude-sonnet-4-5-20250929",
        max_tokens=2000,
        messages=[
            {"role": "user", "content": prompt}
        ]
    )

    return message.content[0].text

def main():
    # Load the episodes file
    episodes_file = '/home/parker/Projects/givewell-board-meetings/givewell-meetings/episodes.json'

    with open(episodes_file, 'r', encoding='utf-8') as f:
        data = json.load(f)

    # Initialize Anthropic client
    api_key = os.environ.get('ANTHROPIC_API_KEY')
    if not api_key:
        raise ValueError("ANTHROPIC_API_KEY environment variable not set")

    client = anthropic.Anthropic(api_key=api_key)

    total_episodes = len(data['episodes'])
    print(f"Processing {total_episodes} episodes...")

    # Process each episode
    for i, episode in enumerate(data['episodes'], 1):
        episode_id = episode.get('id', 'unknown')
        print(f"\n[{i}/{total_episodes}] Processing episode: {episode_id} - {episode.get('title', 'Unknown')}")

        try:
            # Generate summary
            summary = generate_summary(episode, client)

            # Update the description field
            episode['description'] = summary

            print(f"  ✓ Generated summary ({len(summary)} characters)")

        except Exception as e:
            print(f"  ✗ Error processing episode: {e}")
            continue

    # Save the updated JSON
    with open(episodes_file, 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=2, ensure_ascii=False)

    print(f"\n✓ Complete! Updated {episodes_file}")

if __name__ == '__main__':
    main()

File: givewell-meetings/scripts/generate-summaries.js
#!/usr/bin/env node
/**
 * Generate comprehensive 1000-word summaries for each episode
 * Uses Claude API to generate summaries from meeting documents
 */

const fs = require('fs');
const path = require('path');

// Read episodes.json
const episodesPath = path.join(__dirname, '..', 'episodes.json');
const data = JSON.parse(fs.readFileSync(episodesPath, 'utf8'));

// Function to generate summary using Claude (you'll need to provide API key)
async function generateSummary(episode) {
  // Combine all attachment texts
  const allText = episode.attachments
    .map(att => `\n\n=== ${att.title} ===\n${att.text}`)
    .join('\n');

  const prompt = `You are analyzing a board meeting for GiveWell (originally called "The Clear Fund").

Meeting: ${episode.title}
Date: ${episode.dateDisplay}

Here are all the documents from this meeting:
${allText}

Please write a comprehensive 1000-word summary that:
1. Covers the key decisions made
2. Explains important discussions and debates
3. Summarizes financial/budget information
4. Notes any strategic direction or policy changes
5. Highlights personnel or organizational changes
6. Mentions any votes taken and their outcomes

Write this as a professional brief that someone could read to understand everything important that happened in this meeting. Focus on substance over process. Be specific with numbers, names, and decisions.

Write EXACTLY around 1000 words - not much shorter or longer.`;

  console.log(`\n=== Generating summary for Episode ${episode.meetingNumber}: ${episode.title} ===`);
  console.log(`Prompt length: ${prompt.length} characters`);
  console.log(`Source text length: ${allText.length} characters`);

  // TODO: Replace this with actual Claude API call
  // For now, return a placeholder that you'll need to replace
  return `[Summary for ${episode.title} - ${episode.dateDisplay}]\n\nThis is a placeholder. You need to:\n1. Add Claude API integration\n2. Or manually generate summaries\n3. Or use another AI service`;
}

async function main() {
  console.log(`Found ${data.episodes.length} episodes to process\n`);

  // Process each episode
  for (let i = 0; i < data.episodes.length; i++) {
    const episode = data.episodes[i];

    if (episode.attachments.length === 0) {
      console.log(`Skipping episode ${episode.meetingNumber} - no attachments`);
      continue;
    }

    const summary = await generateSummary(episode);

    // Update the episode description
    episode.description = summary;

    console.log(`✓ Generated summary (${summary.length} characters)`);

    // Save progress after each episode
    fs.writeFileSync(episodesPath, JSON.stringify(data, null, 2));
    console.log(`✓ Saved to episodes.json`);
  }

  console.log('\n=== All summaries generated! ===');
}

// Check if this is being run directly
if (require.main === module) {
  main().catch(console.error);
}

module.exports = { generateSummary };
